### PR TITLE
Use concat instead of +=

### PR DESCRIPTION
**Pull request by matthu017**

### CODE REVIEW

**Comment by matthu017** [comment]
Issue: https://github.com/OSC/ood_core/issues/157

File: lib/ood_core/job/adapters/lsf/helper.rb [hunk 1]
```diff
@@ -78,40 +78,40 @@ def parse_cpu_used(cpu_used)
...
-    args += ["-sp", script.priority] unless script.priority.nil?
-    args += ["-H"] if script.submit_as_hold
-    args += (script.rerunnable ? ["-r"] : ["-rn"]) unless script.rerunnable.nil?
-    args += ["-b", script.start_time.localtime.strftime("%Y:%m:%d:%H:%M")] unless script.start_time.nil?
-    args += ["-W", (script.wall_time / 60).to_i] unless script.wall_time.nil?
-    args += ["-L", script.shell_path.to_s] unless script.shell_path.nil?
+    args.concat ["-P", script.accounting_id] unless script.accounting_id.nil?
+    args.concat ["-cwd", script.workdir.to_s] unless script.workdir.nil?
+    args.concat ["-J", script.job_name] unless script.job_name.nil?
+    args[-1].concat "[#{script.job_array_request}]" unless script.job_array_request.nil?
```
Comment (johrstrom) [inline lib/ood_core/job/adapters/lsf/helper.rb:84]: This doesn't work and unfortunately there was no test case to find it. Can you add it in this PR? Specifically that the -J option is modified correctly when given `job_array_request`.

```ruby
irb(main):012:0> args = ['foo']
=> ["foo"]
irb(main):013:0> args[-1] += 'bar'
=> "foobar"
irb(main):014:0> args[-1].concat ['bar']
Traceback (most recent call last):
        3: from /opt/rh/rh-ruby25/root/usr/bin/irb:11:in `<main>'
        2: from (irb):14
        1: from (irb):14:in `concat'
TypeError (no implicit conversion of Array into String)
```

File: lib/ood_core/job/adapters/torque.rb [hunk 2]
```diff
@@ -85,9 +85,9 @@ def submit(script, after: [], afterok: [], afternotok: [], afterany: [])
           depend << "afterany:#{afterany.join(':')}"     unless afterany.empty?
 
           # Set mailing options
-          mail_points  = ""
-          mail_points += "b" if script.email_on_started
-          mail_points += "e" if script.email_on_terminated
+          mail_points = ""
+          mail_points.concat "b" if script.email_on_started
+          mail_points.concat "e" if script.email_on_terminated
```
Comment (ericfranz) [inline lib/ood_core/job/adapters/torque.rb:90]: Since mailpoints is a string, I'd revert this to just using `+=`

**Comment by matthu017** [comment]
This PR has been replaced by https://github.com/OSC/ood_core/pull/203

**Pull request closed without merging**